astroid/interpreter/_import/util.py

```diff
@@ -52,8 +52,11 @@ def is_namespace(modname: str) -> bool:
                 # Check first fragment of modname, e.g. "astroid", not "astroid.interpreter"
                 # because of cffi's behavior
                 # See: https://github.com/PyCQA/astroid/issues/1776
+                mod = sys.modules[processed_components[0]]
                 return (
-                    sys.modules[processed_components[0]].__spec__ is None
+                    mod.__spec__ is None
```

Comment:
I wonder if we even need to check `mod.__spec__ is None`, but I don't fully understand all different styles of namespace packages. The test does pass even if we remove this.